safemysql.class.php

```diff
@@ -170,6 +174,39 @@ public function query()
 		return $this->rawQuery($this->prepareQuery(func_get_args()));
 	}
 
+
+	/**
+	 * Run the provided callback inside a transaction
+	 * @param  callable $callback
+	 * @param  [boolean] $can_retry If the whole transaction should be retried, if it fails due to a deadlock
+	 * @return mixed whatever the callback returns
+	 */
+	public function transaction($callback, $can_retry = false)
+	{
+		$this->query('START TRANSACTION');
+
+		// If there is a deadlock it may be permissable to rerun the entire callback, but individual statements
+		// within it should not be rerun
+		$retry_on_deadlock_status = $this->retryOnDeadlock;
+		$this->retryOnDeadlock = false;
+
+    try
+    {
+    	$result = $this->retryIfDeadlocked($callback, $can_retry);
+	    $this->query('COMMIT');
+	    return $result;
+    } catch (Throwable $e)
+    {
+      $this->query('ROLLBACK');
+      throw $e;
+    } finally
```

Comment:
This bracing style is weird. The library is currently inconsistent between doing

```
if (foo) {
  ...
} else {
  ...
}
```

and

```
if (foo)
{

}
else
{

}
```

but having the `finally` and `catch` here on the same line as the preceding brace but a different line to the following one isn't consistent with how any of the control structures already here are written, unless I've missed something. Your `else` block in `retryIfDeadlocked` is likewise weird.